modules/grpc/bigquery/bigquery-dest.cpp

```diff
@@ -184,6 +234,68 @@ DestinationDriver::construct_schema_prototype()
   this->schema_prototype = msg_factory.GetPrototype(this->schema_descriptor);
 }
 
+bool
+DestinationDriver::load_protobuf_schema()
+{
+  this->protobuf_schema.src_tree = std::make_unique<google::protobuf::compiler::DiskSourceTree>();
+  this->protobuf_schema.src_tree->MapPath(this->protobuf_schema.proto_path, this->protobuf_schema.proto_path);
+
+  this->protobuf_schema.error_coll = std::make_unique<ErrorCollector>();
+
+  this->protobuf_schema.importer =
+    std::make_unique<google::protobuf::compiler::Importer>(this->protobuf_schema.src_tree.get(),
+                                                           this->protobuf_schema.error_coll.get());
+
+  const google::protobuf::FileDescriptor *file_descriptor =
+    this->protobuf_schema.importer->Import(this->protobuf_schema.proto_path);
+
+  if (!file_descriptor || file_descriptor->message_type_count() == 0)
+    {
+      msg_error("Error initializing BigQuery destination, protobuf-schema() file can't be loaded",
+                log_pipe_location_tag(&this->super->super.super.super.super));
+      return false;
+    }
+
+  this->schema_descriptor = file_descriptor->message_type(0);
+
+  this->fields.clear();
+
+  GList *current_value = this->protobuf_schema.values;
+  for (int i = 0; i < this->schema_descriptor->field_count(); ++i)
+    {
+      auto field = this->schema_descriptor->field(i);
+
+      if (!current_value)
+        {
+          msg_error("Error initializing BigQuery destination, protobuf-schema() file has more fields than "
+                    "values listed in the config",
+                    log_pipe_location_tag(&this->super->super.super.super.super));
+          return false;
+        }
+
+      LogTemplate *value = (LogTemplate *) current_value->data;
+
+      this->fields.push_back(Field{field->name(), (google::protobuf::FieldDescriptorProto::Type) field->type(), value});
```

Comment:
We could check whether we support the field type (custom `Message`s are not supported, neither repeated fields, etc...)